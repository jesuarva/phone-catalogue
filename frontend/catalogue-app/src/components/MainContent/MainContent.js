import React from 'react';
import { Switch, Route, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import PhoneListContainer from '../PhoneListContainer/PhoneListContainer';
import PhoneDetailsComponent from '../PhoneDetailComponent/PhoneDetailsComponent';

const MainContent = (props) => {
  const { fetched_Item } = props;
  const { pathname } = props.location;
  console.log(props.location.pathname);
  return (
    <Switch>
      <Route exact path="/" component={PhoneListContainer} />
      <Route path="/catalogue/:model" component={PhoneDetailsComponent} />
    </Switch>
  );
};

const mapStateToProps = (state, dispatch) => {
  return { fetched_Item: state.fetched_Item, dispatch };
};

export default withRouter(
  connect(
    mapStateToProps,
    {},
  )(MainContent),
);
