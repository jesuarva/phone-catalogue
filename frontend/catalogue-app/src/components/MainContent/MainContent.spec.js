import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import mainReducer from '../../reducers/index';
import TestWrapper from '../TestWrapper/TestWrapper';
import MainContent from './MainContent';

configure({ adapter: new Adapter() });

describe('MainContent component', () => {
  const store = createStore(mainReducer, applyMiddleware(thunk));
  const component = (
    <Provider store={store}>
      <Router>
        <MainContent />
      </Router>
    </Provider>
  );

  it('should renders correctly', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('Renders children components', () => {
    const mounted = mount(<TestWrapper>{() => component}</TestWrapper>);

    it('should renders a Switch component', () => {
      const switchComponent = mounted.find(Switch);
      expect(switchComponent).toHaveLength(1);
    });

    it('should renders a Route components', () => {
      const routes = mounted.find(Route);
      expect(routes).toHaveLength(1);
    });
  });
});
