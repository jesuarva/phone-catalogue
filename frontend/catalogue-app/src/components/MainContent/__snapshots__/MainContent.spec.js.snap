// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContent component should renders correctly 1`] = `
<div
  className="row"
/>
`;
